### ISSUE TITLE
Support for tablets/mobile

### ISSUE THREAD

**Issue by bkamins** [bkamins/SilverDecisions#18]
In the future we could consider support for mobile/tablets. I tried using it on a tablet and it would be usable but now it is not possible to do a right-click on a node (text selection is toogled).

**Comment by bkamins** [comment]
@mwasiluk Just an idea (so that I remember it when we come back to it): maybe we can have SilverDecisions reader for mobile/tablets that would open trees but not allow to edit it.

**Comment by mwasiluk** [comment]
We have already readOnly mode implemented (for SPOD datalets).

I'll describe available app initial configuration in the wiki, maybe on the separate page named 'Installation'?

**Comment by bkamins** [comment]
For me having only "Developer's guide" and "SilverDecisions in SPOD" wiki pages is enough (and this functionality probably fits Developer's guide section + some mention in page about SPOD integration).
This is not urgent.

**Comment by bkamins** [comment]
Depends on #59.
We either close it or change overall SilverDecisions behavior.
Only Chrome is planned to be supported on Android.

@michaljakubczyk what should be supported on iPhone?

**Comment by bkamins** [comment]
@annawier when we document tablet/mobile functionality please test on which apps it works correctly (it seems that Chrome is 100% ok except right-mouse click which we will change in relase 0.4).

**Comment by bkamins** [comment]
Currently: on tablet double click is long press and double tapping zooms; single tapping selects - and this works correctly.

@mwasiluk After e7c61f0:
1) when canvas is empty long press correctly opens context menu and allows adding of a new node or text field
2) when there is anything in canvas long press selects it (as if you have pressed Ctrl-A to select all content in canvas). Can this be fixed? (I do not care if we handle long-press correctly also when something is in canvas or handle double tap which is currently zoom as double click)

**Comment by bkamins** [comment]
@mwasiluk how it is supposed to work so that we know what to expect when testing and how to document it.

@annawier please document it.

**Comment by mwasiluk** [comment]
context menu should appear on long press

**Comment by bkamins** [comment]
@michaljakubczyk @annawier I have tested it on Android/Chrome and seems to work properly. Before we close it please provide more test results (in particular - on what os/browser combinations it works).

**Comment by annawier** [comment]
SD works non my phone (Android, chrome) too:
- double click --> zoom (zoom scale by double click  on the left panel > zoom scale by double click on the canvas) shall I document that difference too or it's irrelevant? 
- long press --> context menu
What's not working:
- multitouching on the left panel zooms, but it doesn't work on the canvas
- scrolling works on the left panel too but not on the canvas. So for expanded diagrams it might be not possible to display a whole diagram in zoom
- I can't save it to json, although png and svg works properly

All the information above should be documented?

On the defaulted browser on Android just scrolling and multitouching works, toolbar is not displayed correctly and it is not possible to ass any nodes.

**Comment by bkamins** [comment]
@mwasiluk please comment what is doable and what is problematic here. Mobile is not a priority, but if we can correct those issues cheaply I would like to have them.

**Comment by mwasiluk** [comment]
panning/scrolling and multitouch zooming on canvas collide with selection brush. 
https://github.com/d3/d3/issues/1806

The one thing i've come up with is disabling brush for a few seconds after multitouch pinch is detected

**Comment by bkamins** [comment]
@annawier  Please test this reasonably fast as we want to go with 0.4 to production. I have published current dev on http://bogumilkaminski.pl/sddev/.
Then please update the documentation accordingly. If you feel that text in About dialog should be updated please post here an appropriate message.

@mwasiluk I have tested it on chrome/android and zooming and scrolling works. The only thing that you can zoom in but not zoom out (to make the tree smaller so that it fits the screen).

@mwasiluk The open question is save to JSON. If it will not be possible please say so and @annawier will document it in `Known issues` section.

We have one small bug on mobile (if it cannot be fixed we can live with it - and @annawier will document it in `Known issues` section):
1) select a node
2) perform some action from context menu (eg. copy subtree)
3) select another node

on the original node the dashed line remains around (and should become solid)

**Comment by bkamins** [comment]
Both fixed but another minor issue.

If you select a terminal node of a tree everything is ok.
If you select a non-terminal node it is selected but zoom on this node is done (the node is enlarged).

**Comment by annawier** [comment]
Zoom and scrolling on the canvas works, I can save to json. I've tested other functions and all of them seem to work properly.

Note: there is a warning in the top **right** corner saying that "selection brush enabled/disabled". But this warning is not visible if one zooms at the decision tree, which is always aligned to the **left** top corner. I didn't notice the warning at first, because it was out of sight, so maybe it would be a good idea to place it in the top **left** corner of the canvas?
I mean that sometimes it is allowed to scroll and zoom on the canvas and sometimes not: the same situation is with selecting nodes. If one doesn't notice the warning immediately because it's completely hidden, it might be quite confusing (please see the pics).
![whatsapp image 2016-12-17 at 21 37 12](https://cloud.githubusercontent.com/assets/23300430/21289613/9f6cfa28-c4a1-11e6-9a3f-18c7e382ff7d.jpeg)

**Comment by bkamins** [comment]
@mwasiluk I understand that moving this pop-up should not be a problem?

**Comment by bkamins** [comment]
@annawier we are left with documentation

**Comment by annawier** [comment]
It's already in dev branch, I'll check it (especially formatting) when it's
published :)

2016-12-19 13:25 GMT+01:00 Bogumił Kamiński <notifications@github.com>:

**Comment by bkamins** [comment]
@michaljakubczyk reports problems on iPad

**Comment by bkamins** [comment]
iPad will not be supported
